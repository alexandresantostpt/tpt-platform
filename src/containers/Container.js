import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'

const StyledContainer = styled.main`
    background: #fbfbfb;
    border: 1px solid #dcdcdc;
    grid-area: container;
    overflow-y: auto;
    padding: 3.125rem 0;
`

const Container = ({ children }) => <StyledContainer>{children}</StyledContainer>
Container.defaultProps = {
    children: null
}

Container.propTypes = {
    children: PropTypes.oneOfType([PropTypes.array, PropTypes.bool, PropTypes.number, PropTypes.object, PropTypes.string])
}

export default Container
