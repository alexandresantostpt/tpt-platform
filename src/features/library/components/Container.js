import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'

import BgLibrary from '@img/background/library.png'

const StyledContainer = styled.main`
    background: #fbfbfb;
    background-image: ${({ showBg }) => (showBg ? `url(${BgLibrary})` : null)};
    background-position: center;
    background-repeat: no-repeat;
    background-size: cover;
    border: 1px solid #dcdcdc;
    grid-area: container;
    padding: 3.125rem 0;
`

const Container = ({ children, showBg }) => <StyledContainer showBg={showBg}>{children}</StyledContainer>
Container.defaultProps = {
    children: null,
    showBg: false
}

Container.propTypes = {
    children: PropTypes.oneOfType([PropTypes.array, PropTypes.bool, PropTypes.number, PropTypes.object, PropTypes.string]),
    showBg: PropTypes.bool
}

export default Container
