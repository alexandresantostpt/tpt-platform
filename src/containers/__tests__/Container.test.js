import React from 'react'
import renderer from 'react-test-renderer'

import Container from '../Container'

describe('Tests for Container container', () => {
    it('Should rendering without crash ', () => {
        const tree = renderer
            .create(
                <Container>
                    <h1>Must be show</h1>
                </Container>
            )
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})
