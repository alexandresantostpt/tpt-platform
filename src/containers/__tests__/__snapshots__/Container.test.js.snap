// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Container container Should rendering without crash  1`] = `
<main
  className="Container__StyledContainer-sc-1i8buc6-0 SsRsD"
>
  <h1>
    Must be show
  </h1>
</main>
`;
